Sources/Widgets/Widgets3D/ResliceCursorWidget/behavior.js

```diff
@@ -83,36 +85,93 @@ export default function widgetBehavior(publicAPI, model) {
     const image = model.widgetState.getImage();
     const imageSpacing = image.getSpacing();
     const imageBounds = image.getBounds();
-    const boundingBox = vtkBoundingBox.newInstance({ bounds: imageBounds });
+    const bounds = vtkBoundingBox.newInstance({ bounds: imageBounds });
 
     // Get the normal use to translate the center along
-    const normal = calldata.pokedRenderer
+    const dirProj = calldata.pokedRenderer
       .getRenderWindow()
       .getRenderers()[0]
       .getActiveCamera()
       .getDirectionOfProjection();
 
-    // Direction of the projection if the inverse of what we want
-    const translationVector = vtkMath.multiplyScalar(normal, -1);
+    // Direction of the projection is the inverse of what we want
+    const normal = vtkMath.multiplyScalar(dirProj, -1);
+
+    const XPlaneNormal = model.widgetState.getXPlaneNormal();
+    const YPlaneNormal = model.widgetState.getYPlaneNormal();
+    const ZPlaneNormal = model.widgetState.getZPlaneNormal();
+
+    // Find in which view we are
+    let otherViews = ['X', 'Y'];
+    let otherNormals = [XPlaneNormal, YPlaneNormal];
+    if (vtkMath.dot(normal, XPlaneNormal) > 0.9) {
+      otherViews = ['Y', 'Z'];
+      otherNormals = [YPlaneNormal, ZPlaneNormal];
+    } else if (vtkMath.dot(normal, YPlaneNormal) > 0.9) {
+      otherViews = ['X', 'Z'];
+      otherNormals = [XPlaneNormal, ZPlaneNormal];
+    }
 
+    // Define the potentially new center
     let newCenter = [
-      currentCenter[0] + direction * translationVector[0] * imageSpacing[0],
-      currentCenter[1] + direction * translationVector[1] * imageSpacing[1],
-      currentCenter[2] + direction * translationVector[2] * imageSpacing[2],
+      currentCenter[0] + direction * normal[0] * imageSpacing[0],
+      currentCenter[1] + direction * normal[1] * imageSpacing[1],
+      currentCenter[2] + direction * normal[2] * imageSpacing[2],
     ];
-
-    if (!boundingBox.containsPoint(...newCenter)) {
-      // Clamp the center to the bounds of the image
-      const imageOrigin = image.getOrigin();
-      const localRoundedCenter = currentCenter.map(
-        (value, index) =>
-          Math.round(value - imageOrigin[index]) / imageSpacing[index]
-      );
-
-      // Get it back in world coordinates
-      newCenter = localRoundedCenter.map(
-        (value, index) => value * imageSpacing[index] + imageOrigin[index]
-      );
+    const newCenterInsideBounds = bounds.containsPoint(...newCenter);
+
+    // The center is out of the image
+    if (!newCenterInsideBounds) {
+      // Find which plane is out of the bounds
+      const isCurrentPlaneOutOfBounds =
```

Comment:
isn't it `isNewPlaneOutOfBounds` according to the following 2 lines ?

Reply:
The variable tests if the current position is correct and the next one, out of the bounds. We can call it both.
Current only means here that it's the current view type (for exemple X if we are scrolling on X view)

Reply:
here "current" means too things: `currentCenter` vs `newCenter` and current as in currentView. 
Maybe you could rename `currentCenter` by `oldCenter` to avoid confusion.